### PR TITLE
util/pingpong: Add support of pingpong and strerror smoke tests on WinOS

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
1) Add strerror and pingpong VS projects
2) Update Win, Unix OS dependency files
3) Update pingpong.c test

### CODE REVIEW

File: util/pingpong.c [hunk 1]
```diff
@@ -63,10 +64,10 @@
...
 #endif
 
-#ifdef __APPLE__
-#include "osx/osd.h"
-#elif defined __FreeBSD__
-#include "freebsd/osd.h"
+#ifdef WIN32
+#define PP_TRY_AGAIN WSAETIMEDOUT
+#else
+#define PP_TRY_AGAIN EAGAIN
```
Comment (shefty) [inline util/pingpong.c]: Please move these defines be moved to the corresponding osd.h files.

**Comment by gladkovdmitry17** [comment]
Sure, sound reasonable
This macro is needed only in case when socket option SO_RCVTIMEO is set and we don't use polling, i.e. just trying to receive data in loop ignoring case when EAGAIN/WSAETIMEDOUT error is occurred.
Is it OK to hardcode such constant? If yes, OFI_TRY_RCV_AGAIN is ok? Just to highlight that this macro must be used only in case of recv(). Not sure what is returned from connect() in case of timed out on WinOS (just need to check)

--------------------------------------------------------------------
Joint Stock Company Intel A/O
Registered legal address: Krylatsky Hills Business Park,
17 Krylatskaya Str., Bldg 4, Moscow 121614,
Russian Federation

This e-mail and any attachments may contain confidential material for
the sole use of the intended recipient(s). Any review or distribution
by others is strictly prohibited. If you are not the intended
recipient, please contact the sender and delete all copies.

**Comment by gladkovdmitry17** [comment]
One more thing, I need to update logging macro to support both type of errno (Unix- and Win-specific), since I've replaced errno to common ofi_socket_err()

**Comment by shefty** [comment]
@jsquyres @bturrubiates - please see the changes to the pingpong test to support windows.

**Comment by bturrubiates** [comment]
> @jsquyres @bturrubiates - please see the changes to the pingpong test to support windows.

Haven't taken a full look yet, but I see a lot of socket changes. We were planning on removing all of the socket functionality from the test soon.

**Comment by shefty** [comment]
Are you okay removing it after these changes, so that they can at least run this on windows?  I just want to avoid conflicts against pending changes at this point.

**Comment by bturrubiates** [comment]
> Are you okay removing it after these changes, so that they can at least run this on windows?  I just want to avoid conflicts against pending changes at this point.

Sure, I'm fine with that.

**Pull request merged**